connectors/sources/zoom.py

```diff
@@ -0,0 +1,540 @@
+
+        if cached_value:
+            return cached_value
+
+        now = datetime.utcnow()
+        access_token, expires_in = await self._fetch_token()
+        self._token_cache.set_value(access_token, now + timedelta(seconds=expires_in))
+
+        return access_token
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def _fetch_token(self):
+        self._logger.debug("Generating access token.")
+        url = AUTH.format(account_id=self.account_id)
+        content = f"{self.client_id}:{self.client_secret}"
+        base64_credentials = get_base64_value(content=content.encode("utf-8"))
+        request_headers = {
+            "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
+            "Authorization": f"Basic {base64_credentials}",
+        }
+
+        try:
+            async with self._http_session.post(
+                url=url, headers=request_headers
+            ) as response:
+                json_response = await response.json()
+                return json_response.get("access_token"), json_response.get(
+                    "expires_in", 3599
+                )
+        except Exception as exception:
+            raise TokenError(
+                f"Error while generating access token. Exception {exception}."
+            ) from exception
+
+
+class ZoomAPISession:
+    def __init__(self, http_session, api_token, logger_):
+        self._sleeps = CancellableSleeps()
+        self._logger = logger_
+
+        self._http_session = http_session
+        self._api_token = api_token
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+
+    def close(self):
+        self._sleeps.cancel()
+
+    @asynccontextmanager
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+        skipped_exceptions=NotFound,
+    )
+    async def _get(self, absolute_url):
+        try:
+            token = await self._api_token.get()
+            headers = {
+                "authorization": f"Bearer {token}",
+                "content-type": "application/json",
+            }
+
+            async with self._http_session.get(
+                url=absolute_url, headers=headers
+            ) as response:
+                yield response
+        except ClientResponseError as exception:
+            if exception.status == 401:
+                await self._api_token.get(is_cache=False)
+                raise
+            elif exception.status == 404:
+                raise NotFound from exception
+            else:
+                raise
+
+    async def fetch(self, url):
+        try:
+            async with self._get(absolute_url=url) as response:
+                return await response.json()
+        except Exception:
+            self._logger.warning(f"Data for {url} is being skipped.")
```

Comment:
Should probably log why (check to see if the exception has a message, and at least log the class, maybe debug-log the stack trace)